File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        
        
        StringBuilder sb = new StringBuilder();
        int minWord = Integer.MAX_VALUE;
        
        for(String str : strs){
            minWord = Math.min(minWord, str.length());
        }
        
        
        for(int i = 0; i < minWord; i++){
            
            for(int j = 0; j < strs.length; j++){
                if(strs[0].charAt(i) != strs[j].charAt(i)) {
                    return sb.toString();
                }
            }
            sb.append(strs[0].charAt(i));
        }
        
        //["ab", "a"]
        return sb.toString();
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {

        if strs.is_empty() {
            return String::new();
        }

        let shortest_word = strs.iter().map(|w| w.len()).min().unwrap();

        let first_word = strs[0].as_bytes();

        for i in 0..shortest_word {

            let c = first_word[i];

            for word in strs.iter().skip(1) {

                if c != word.as_bytes()[i] {
                    return strs[0][..i].to_string();
                }
            }
        }

        return strs[0][..shortest_word].to_string();

    }
}